test/datasource/pypi.spec.js

```diff
@@ -39,6 +39,14 @@ describe('datasource/pypi', () => {
       await datasource.getDependency('pkg:pypi/azure-cli-monitor', config);
       expect(got.mock.calls).toMatchSnapshot();
     });
+    it('supports custom datasource url from environmental variable', async () => {
+      got.mockReturnValueOnce({
+        body: JSON.parse(res1),
+      });
+      process.env.PIP_INDEX_URL = 'https://my.pypi.python/pypi/'
```

Comment:
Does this leak to other tests?

Reply:
I suspect process.env is per-file however it would be good to save and replace here just to be safe.

Reply:
By save and replace do you mean to save the value of PIP_INDEX_URL to a variable before setting it and then set the value back after the test; or do you mean unset the variable altogether after the test?

Reply:
Probably either is OK, but the first one is most foolproof